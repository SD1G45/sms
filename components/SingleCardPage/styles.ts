import styled from "styled-components";

export const Page = styled.div`
  width: 100%;
  height: 100vh;
  display: flex;
  justify-content: center;
  align-items: center;
`;

export const LeftTriangleDiv = styled.div`
  padding-top: 301px;
  margin-right: 20px;
`;

export const RightCircleDiv = styled.div`
  padding-bottom: 301px;
  margin-top: 100px;
  margin-left: 20px;
`;
