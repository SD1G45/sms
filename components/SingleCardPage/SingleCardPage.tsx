import React from "react";
import Background from "../Background";
import Image from "next/image";
import { LeftTriangleDiv, Page, RightCircleDiv } from "./styles";
import { Parallax } from "react-scroll-parallax";

const SingleCardPage: React.FC = ({ children }) => {
  return (
    <Page>
      <Background />
      <Parallax speed={2}>
        <LeftTriangleDiv>
          <Image
            src="/StackedTriangle-1.svg"
            width={200}
            height={300}
            alt="triangle"
          />
        </LeftTriangleDiv>
      </Parallax>
      {children}

      <Parallax speed={110}>
        <RightCircleDiv>
          <Image
            src="/StackedCircle.svg"
            width={300}
            height={300}
            alt="circle"
          />
        </RightCircleDiv>
      </Parallax>
    </Page>
  );
};

export default SingleCardPage;
