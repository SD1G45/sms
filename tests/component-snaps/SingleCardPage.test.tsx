import React from "react";
import renderer from "react-test-renderer";
import "jest-styled-components";
import SingleCardPage from "../../components/SingleCardPage";

describe("<SingleCardPage/>", () => {
  it("renders correctly", () => {
    const tree = renderer.create(<SingleCardPage/>).toJSON();
    expect(tree).toMatchSnapshot();
  });
});