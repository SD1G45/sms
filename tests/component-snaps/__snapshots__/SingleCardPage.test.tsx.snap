// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<SingleCardPage/> renders correctly 1`] = `
.c3 {
  height: 100vh;
  width: 260px;
}

.c4 {
  height: 100vh;
  width: 250px;
  border-left: 1px dashed #ccc;
}

.c2 {
  height: 100vh;
  width: 230vw;
  position: fixed;
  bottom: 55vh;
  left: -60vw;
  background-color: #f7f8ff;
  -webkit-transform: rotate(165deg);
  -ms-transform: rotate(165deg);
  transform: rotate(165deg);
  z-index: -2;
}

.c1 {
  position: absolute;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-box-pack: justify;
  -webkit-justify-content: space-between;
  -ms-flex-pack: justify;
  justify-content: space-between;
  width: 100%;
  z-index: -1;
}

.c0 {
  width: 100%;
  height: 100vh;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-box-pack: center;
  -webkit-justify-content: center;
  -ms-flex-pack: center;
  justify-content: center;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
}

.c5 {
  padding-top: 301px;
  margin-right: 20px;
}

.c6 {
  padding-bottom: 301px;
  margin-top: 100px;
  margin-left: 20px;
}

<div
  className="c0"
>
  <div
    className="c1"
  >
    <div
      className="c2"
    />
    <div
      className="c3"
    />
    <div
      className="c4"
    />
    <div
      className="c4"
    />
    <div
      className="c4"
    />
    <div
      className="c4"
    />
    <div
      className="c4"
    />
    <div
      className="c4"
    />
    <div
      className="c4"
    />
  </div>
  <div
    className="c5"
  >
    <div
      style={
        Object {
          "boxSizing": "border-box",
          "display": "inline-block",
          "margin": 0,
          "maxWidth": "100%",
          "overflow": "hidden",
          "position": "relative",
        }
      }
    >
      <div
        style={
          Object {
            "boxSizing": "border-box",
            "display": "block",
            "maxWidth": "100%",
          }
        }
      >
        <img
          alt=""
          aria-hidden={true}
          role="presentation"
          src="data:image/svg+xml;base64,PHN2ZyB3aWR0aD0iMjAwIiBoZWlnaHQ9IjMwMCIgeG1sbnM9Imh0dHA6Ly93d3cudzMub3JnLzIwMDAvc3ZnIiB2ZXJzaW9uPSIxLjEiLz4="
          style={
            Object {
              "border": "none",
              "display": "block",
              "margin": 0,
              "maxWidth": "100%",
              "padding": 0,
            }
          }
        />
      </div>
      <noscript>
        <img
          alt="triangle"
          decoding="async"
          src="/StackedTriangle-1.svg"
          srcSet="/_next/image?url=%2FStackedTriangle-1.svg&w=256&q=75 1x, /_next/image?url=%2FStackedTriangle-1.svg&w=640&q=75 2x"
          style={
            Object {
              "border": "none",
              "bottom": 0,
              "boxSizing": "border-box",
              "display": "block",
              "height": 0,
              "left": 0,
              "margin": "auto",
              "maxHeight": "100%",
              "maxWidth": "100%",
              "minHeight": "100%",
              "minWidth": "100%",
              "objectFit": undefined,
              "objectPosition": undefined,
              "padding": 0,
              "position": "absolute",
              "right": 0,
              "top": 0,
              "width": 0,
            }
          }
        />
      </noscript>
      <img
        alt="triangle"
        decoding="async"
        src="data:image/gif;base64,R0lGODlhAQABAIAAAAAAAP///yH5BAEAAAAALAAAAAABAAEAAAIBRAA7"
        style={
          Object {
            "border": "none",
            "bottom": 0,
            "boxSizing": "border-box",
            "display": "block",
            "height": 0,
            "left": 0,
            "margin": "auto",
            "maxHeight": "100%",
            "maxWidth": "100%",
            "minHeight": "100%",
            "minWidth": "100%",
            "objectFit": undefined,
            "objectPosition": undefined,
            "padding": 0,
            "position": "absolute",
            "right": 0,
            "top": 0,
            "width": 0,
          }
        }
      />
    </div>
  </div>
  <div
    className="c6"
  >
    <div
      style={
        Object {
          "boxSizing": "border-box",
          "display": "inline-block",
          "margin": 0,
          "maxWidth": "100%",
          "overflow": "hidden",
          "position": "relative",
        }
      }
    >
      <div
        style={
          Object {
            "boxSizing": "border-box",
            "display": "block",
            "maxWidth": "100%",
          }
        }
      >
        <img
          alt=""
          aria-hidden={true}
          role="presentation"
          src="data:image/svg+xml;base64,PHN2ZyB3aWR0aD0iMzAwIiBoZWlnaHQ9IjMwMCIgeG1sbnM9Imh0dHA6Ly93d3cudzMub3JnLzIwMDAvc3ZnIiB2ZXJzaW9uPSIxLjEiLz4="
          style={
            Object {
              "border": "none",
              "display": "block",
              "margin": 0,
              "maxWidth": "100%",
              "padding": 0,
            }
          }
        />
      </div>
      <noscript>
        <img
          alt="circle"
          decoding="async"
          src="/StackedCircle.svg"
          srcSet="/_next/image?url=%2FStackedCircle.svg&w=384&q=75 1x, /_next/image?url=%2FStackedCircle.svg&w=640&q=75 2x"
          style={
            Object {
              "border": "none",
              "bottom": 0,
              "boxSizing": "border-box",
              "display": "block",
              "height": 0,
              "left": 0,
              "margin": "auto",
              "maxHeight": "100%",
              "maxWidth": "100%",
              "minHeight": "100%",
              "minWidth": "100%",
              "objectFit": undefined,
              "objectPosition": undefined,
              "padding": 0,
              "position": "absolute",
              "right": 0,
              "top": 0,
              "width": 0,
            }
          }
        />
      </noscript>
      <img
        alt="circle"
        decoding="async"
        src="data:image/gif;base64,R0lGODlhAQABAIAAAAAAAP///yH5BAEAAAAALAAAAAABAAEAAAIBRAA7"
        style={
          Object {
            "border": "none",
            "bottom": 0,
            "boxSizing": "border-box",
            "display": "block",
            "height": 0,
            "left": 0,
            "margin": "auto",
            "maxHeight": "100%",
            "maxWidth": "100%",
            "minHeight": "100%",
            "minWidth": "100%",
            "objectFit": undefined,
            "objectPosition": undefined,
            "padding": 0,
            "position": "absolute",
            "right": 0,
            "top": 0,
            "width": 0,
          }
        }
      />
    </div>
  </div>
</div>
`;
